src/audio/host-zephyr.c

```diff
@@ -912,23 +920,24 @@ int host_common_params(struct host_data *hd, struct comp_dev *dev,
 			buffer_addr = addr;
 	}
 
-	dma_block_cfg.block_size = buffer_bytes;
+	dma_block_cfg->block_size = buffer_bytes;
 
 	switch (config->direction) {
 	case DMA_DIR_LMEM_TO_HMEM:
 		dma_cfg->channel_direction = MEMORY_TO_HOST;
-		dma_block_cfg.source_address = buffer_addr;
+		dma_block_cfg->source_address = buffer_addr;
 		break;
 	case DMA_DIR_HMEM_TO_LMEM:
 		dma_cfg->channel_direction = HOST_TO_MEMORY;
-		dma_block_cfg.dest_address = buffer_addr;
+		dma_block_cfg->dest_address = buffer_addr;
 		break;
 	}
 
 	err = dma_config(hd->chan->dma->z_dev, hd->chan->index, dma_cfg);
 	if (err < 0) {
 		comp_err(dev, "host_params(): dma_config() failed");
 		dma_release_channel(hd->dma->z_dev, hd->chan->index);
+		rfree(dma_block_cfg);
```

Comment:
do we also want to reset `dma_cfg->head_block` to `NULL` before freeing? Might be worth making an error handling label at the bottom of the function

Reply:
fixed, should be ok now